domains/client/domain-operator/src/domain_block_processor.rs

```diff
@@ -419,6 +420,15 @@ where
         inherent_digests: Digest,
         inherent_data: sp_inherents::InherentData,
     ) -> Result<DomainBlockBuildResult<Block>, sp_blockchain::Error> {
+        // TODO: This is used to keep compatible with gemini-3h, remove before next network
+        let is_gemini_3h = self.consensus_client.info().genesis_hash
+            == FromStr::from_str(
+                // The genesis hash of gemini-3h
+                "0c121c75f4ef450f40619e1fca9d1e8e7fbabc42c895bc4790801e85d5a91c34",
+            )
+            .map_err(|_| ())
+            .expect("parsing consensus block hash should success");
```

Comment:
Assuming parsing works, this should do the job, but even better way would be to define genesis has as `[u8; 32]` constant and store boolean result as a property of the data structure to not query info all the time since genesis hash never changes.